### PR TITLE
Handle newly added and partly modified indexes

### PR DESCRIPTION
**Pull request by tharindu-bandara**

### CODE REVIEW

File: modules/migration/migration-resources/5.8.0/dbscripts/step1/identity/mysql5.7.sql [hunk 1]
```diff
@@ -22,9 +63,54 @@ CREATE INDEX IDX_USER_ID ON IDN_AUTH_USER_SESSION_MAPPING (USER_ID);
 
 CREATE INDEX IDX_SESSION_ID ON IDN_AUTH_USER_SESSION_MAPPING (SESSION_ID);
 
+call create_index_if_not_column_is_partly_indexed('ATTR_NAME', 'IDX_IDN_SCIM_GROUP_TI_RN_AN', 'IDN_SCIM_GROUP', '(TENANT_ID, ROLE_NAME, ATTR_NAME(500))');
```
Comment (omindu) [inline modules/migration/migration-resources/5.8.0/dbscripts/step1/identity/mysql5.7.sql]: Calling functions passing the index names as variables might lead to the below error

```
SQLException: Illegal mix of collations (utf8_general_ci,IMPLICIT) and (utf8_unicode_ci,IMPLICIT) for operation '='
```

Can we first verify whether we get the above with the following DB collation configurations.

`SHOW VARIABLES LIKE 'collation%';`

<table>
    <tbody>
        <tr>
            <td>collation_connection</td>
            <td >utf8_general_ci</td>
        </tr>
        <tr>
            <td>collation_database</td>
            <td >utf8_unicode_ci</td>
        </tr>
        <tr>
            <td>collation_server</td>
            <td >latin1_swedish_ci</td>
        </tr>
    </tbody>
</table>

Similar issue and the fix can be found at https://github.com/wso2/product-is/issues/4191

Comment (tharindu-bandara) [reply modules/migration/migration-resources/5.8.0/dbscripts/step1/identity/mysql5.7.sql]: Tested calling the stored procedure with the above collation settings. Did not run into any errors.

Comment (tharindu-bandara) [reply modules/migration/migration-resources/5.8.0/dbscripts/step1/identity/mysql5.7.sql]: There was a misconfiguration in collation configs during the above testing. Once that is corrected, I was able to reproduce the Illegal mix of collations error. 

This issue is fixed in https://github.com/wso2/product-is/pull/5278.

**Comment by tharindu-bandara** [comment]
Closing as this is addressed in https://github.com/wso2/product-is/pull/5278.

**Pull request closed without merging**